src/settings.c

```diff
@@ -279,6 +386,47 @@ pwquality_read_config(pwquality_settings_t *pwq, const char *cfgfile, void **aux
         return read_config_file(pwq, cfgfile, auxerror);
 }
 
+/* change the default configuration file */
+int
+pwquality_set_config_name(pwquality_settings_t *pwq, int setting,
+        const char *cfgfile)
+{
+        char *dup;
+
+        if (cfgfile == NULL) {
+                switch (setting) {
+                case PWQ_SETTING_BASE_CFGFILE:
+                        cfgfile = PWQUALITY_BASE_CFGFILE;
+                        break;
+                case PWQ_SETTING_DEFAULT_CFGFILE:
+                        cfgfile = PWQUALITY_DEFAULT_CFGFILE;
+                        break;
+                default:
+                        return PWQ_ERROR_UNKNOWN_SETTING;
+                }
+        }
+
+        dup = strdup(cfgfile);
+        if (!dup)
+                return PWQ_ERROR_MEM_ALLOC;
+
+        switch (setting) {
+        case PWQ_SETTING_BASE_CFGFILE:
+                free(pwq->base_cfgfile);
+                pwq->base_cfgfile = dup;
+                break;
+        case PWQ_SETTING_DEFAULT_CFGFILE:
+                free(pwq->default_cfgfile);
+                pwq->default_cfgfile = dup;
+                break;
+        default:
+                free(dup);
+                return PWQ_ERROR_UNKNOWN_SETTING;
+        }
+
+        return 0;
```

Comment:
My idea was something like this:

```suggestion
#define PWQUALITY_BASE_PATH "/usr/lib/security"
#define PWQUALITY_DEFAULT_PATH "/etc/security"
#define PWQUALITY_DEFAULT_NAME "pwquality.conf"
/* change the default configuration file name */
int
pwquality_set_config_name(pwquality_settings_t *pwq,
        const char *cfgname)
{
        char *base_path, *default_path;

        if (cfgname == NULL)
                cfgname = PWQUALITY_DEFAULT_NAME;

        if (asprintf("&base_path, "%s/%s", PWQUALITY_BASE_PATH, cfgname) < 0)
                return PWQ_ERROR_MEM_ALLOC;
        if (asprintf("&default_path, "%s/%s", PWQUALITY_DEFAULT_PATH, cfgname) < 0) {
                free(base_path);
                return PWQ_ERROR_MEM_ALLOC;
        }
        free(pwq->base_cfgfile);
        pwq->base_cfgfile = base_path;
        free(pwq->default_cfgfile);
        pwq->default_cfgfile = default_path;
        return 0;
```

Reply:
I don't think it's a good idea to enforce the use of the `security` directory for the default config files.